tmdsimpy/jax/nlforces/elastic_dry_fric_2d.py

```diff
@@ -329,31 +338,39 @@ def aft(self, U, w, h, Nt=128, tol=1e-7):
         # # If no Grad is needed use:
         # Flocal = _local_aft_jenkins(Uwlocal, pars, u0, tuple(h), Nt, u0h0)[0]
         
+        if calc_grad:
         # Case with gradient and local force
-        dFdUwlocal, Flocal = _local_aft_eldry_grad(Uwlocal, pars, u0, \
-                                                     tuple(h), Nt, u0h0)
-        
-        
+            dFdUwlocal, Flocal = _local_aft_eldry_grad(Uwlocal, pars, u0, \
+                                        tuple(h), Nt, u0h0, self.meso_gap)
+        else:
+            Flocal,_ = _local_aft_eldry(Uwlocal, pars, u0, \
+                                        tuple(h), Nt, u0h0, self.meso_gap)
+            
         #########################
         # Convert AFT to Global Coordinates
         
         # Reshape Flocal
         Flocal = jnp.reshape(Flocal, (Ndnl, Nhc), 'F')
-                
+        
         # Global coordinates        
         Fnl = np.reshape(self.T @ Flocal, (U.shape[0],), 'F')
-        dFnldU = np.kron(np.eye(Nhc), self.T) @ dFdUwlocal[:, :-1] \
-                                                @ np.kron(np.eye(Nhc), self.Q)
         
-        dFnldw = np.reshape(self.T @ \
-                            np.reshape(dFdUwlocal[:, -1], (Ndnl, Nhc)), \
-                            (U.shape[0],), 'F')
-        
-        return Fnl, dFnldU, dFnldw
+        if calc_grad:
+            dFnldU = np.kron(np.eye(Nhc), self.T) @ dFdUwlocal[:, :-1] \
+                                                    @ np.kron(np.eye(Nhc), self.Q)
+            
+            dFnldw = np.reshape(self.T @ \
+                                np.reshape(dFdUwlocal[:, -1], (Ndnl, Nhc)), \
+                                (U.shape[0],), 'F')
+            
+            return Fnl, dFnldU, dFnldw
+        else:
+            return (Fnl,)
     
     
     def local_force_history(self, unlt, unltdot, h, cst, unlth0, max_repeats=2,
                             atol=1e-10, rtol=1e-10):
+
```

Comment:
There should not be a blank line between the function definition line and the comment.